core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSample.java

```diff
@@ -469,4 +469,15 @@ public boolean equals(Object obj) {
   public QcTarget getQcTarget() {
     return QcTarget.Sample;
   }
+
+  @Override
+  public Date getBarcodeDate() {
+    return getReceivedDate();
```

Comment:
Use creation date if received date is null?

Reply:
Fixed.